### PR TITLE
[addons] fix display logic for official/3rd party modules

### PR DESCRIPTION
**Pull request by howie-f**
## Description
hiding all scripts/module-dependencies from the dependency info dialog by default is unnecessary. users should be able to inspect which dependencies get installed along with an add-on beforehand.

- all dependencies including modules show up if the dependency select dialog was opened from the add-on info screen
- when add-on install is triggered from the info screen, only non-optionals and 3rd party modules will open the dependency select dialog
- the advanced setting `showalldependencies` introduced in https://github.com/xbmc/xbmc/pull/18986 is now obsolete and removed.

## What is the effect on users?
ability to inspect all dependencies going to be installed beforehand

## Screenshots (if appropriate):
**dependency select dialog opened from info screen:** 
screenshot00000

**installing the Youtube add-on will not pop up the dialog again, because in this case the optional InputStream helper is the only candidate**

maybe @emveepee can runtime test with a 3rd party script/module dependency?

## Types of change

- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document

### CODE REVIEW

**Comment by howie-f** [comment]
jenkins build this please

**Comment by howie-f** [comment]
jenkins build this please

File: xbmc/addons/gui/GUIDialogAddonInfo.cpp
Comment (howie-f) [inline xbmc/addons/gui/GUIDialogAddonInfo.cpp:792]: else path was missing

File: xbmc/addons/gui/GUIDialogAddonInfo.cpp
Comment (phunkyfish) [inline xbmc/addons/gui/GUIDialogAddonInfo.cpp:778]: This feel off with the empty line between the comment and the line it refers to. Maybe an empty line before the comment and no empty line after?

File: xbmc/addons/gui/GUIDialogAddonInfo.cpp
Comment (phunkyfish) [inline xbmc/addons/gui/GUIDialogAddonInfo.cpp:788]: This feel off with the empty line between the comment and the line it refers to. Maybe an empty line before the comment and no empty line after?

File: xbmc/addons/gui/GUIDialogAddonInfo.cpp
Comment (phunkyfish) [inline xbmc/addons/gui/GUIDialogAddonInfo.cpp:797]: Not included in the change but same here.

**Review by phunkyfish — CHANGES_REQUESTED** [review]
Just the empty lines around the comments is confusing.

File: xbmc/addons/gui/GUIDialogAddonInfo.cpp
Comment (howie-f) [reply xbmc/addons/gui/GUIDialogAddonInfo.cpp]: yes, done for all

**Comment by howie-f** [comment]
jenkins build this please

**Comment by howie-f** [comment]
are more changes required here? i consider this a candidate for a backport because i think introducing `showalldependencies` advanced setting wasn‘t the smartest decision.
@phunkyfish and @DaveTBlake how do you think about this?

**Comment by phunkyfish** [comment]
I would merge this in master and wait for any feedback first. On the backport I'll defer to @DaveTBlake

**Review by phunkyfish — APPROVED** [review]
LGTM

**Comment by howie-f** [comment]
ok, then let‘s see how it goes. thx phunkyfish

**Pull request merged**
